packages/magento-product-configurable/components/ConfigurableProductUrls/ConfigurableProductUrls.tsx

```diff
@@ -0,0 +1,30 @@
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import { useConfigurableOptionsSelection } from '../../hooks/useConfigurableOptionsSelection'
+
+type ConfigurableProductUrlsProps = {
+  url_key?: string | null
+  index?: number
+}
+
+export function ConfigurableProductUrls(props: ConfigurableProductUrlsProps) {
+  const { url_key, index = 0 } = props
+  const router = useRouter()
+  const { configured } = useConfigurableOptionsSelection({ url_key, index })
+
+  useEffect(() => {
+    const currentUrlPath = router.asPath.split('/')
+    const currentUrlKey = currentUrlPath[currentUrlPath.length - 1]
+    const simpleProductUrlKey = configured?.configurable_product_options_selection?.variant?.url_key
+
+    if (simpleProductUrlKey && simpleProductUrlKey !== currentUrlKey) {
+      router
+        .replace(`/p/${simpleProductUrlKey}`, undefined, { scroll: false, shallow: true })
```

Comment:
There is a `productLink` method to get the correct route. I believe some projects are using a different route than `/p/[url]`.